controllers/openshift_controller.go

```diff
@@ -406,6 +399,40 @@ func (r *KataConfigOpenShiftReconciler) checkNodeEligibility() error {
 	return nil
 }
 
+func (r *KataConfigOpenShiftReconciler) getNodeSelectorAsMap() map[string]string {
```

Comment:
Great that you factored this code out to its own function! There's one more place - in newMCPforCR() - that builds a very similar selector, could the relationship between the two selector be used to simplify the code even more?

I realise that the two selectors aren't precisely the same (the one in newMCPforCR() is used to label nodes with 'kata-oc' whereas this one uses 'kata-oc') but still, the amount of code is considerable and the similarities close enough to be worth considering a further refactor - what do you think?

Reply:
You are right. I was thinking about refactoring the code in newMCPforCR as well, however newMCPforCR expects labelselectors. So the method will be something like get NodeSelectorAsSelector(). Hence left it as-is to tackle it later on. Need some investigation to find out existing helper methods for conversion and any limitations.